### DISCUSSION TITLE
how do i connect to LIMA using SSH?

### DISCUSSION THREAD

**Discussion by Glar35** [lima-vm/lima#1221] [Q&A] [answered]
I want to connect to lima (with CLion, from my mac) using ssh,
how can i find the host, username, port and password?
thanks!

**Comment by AkihiroSuda** [comment, marked as answer]
```bash
limactl show-ssh --format=config default > ./ssh-config
ssh -F ./ssh-config lima-default
```

More info:
```console
$ limactl show-ssh --help
Show the ssh command line

Usage:
  limactl show-ssh [flags] INSTANCE

Examples:

  "cmd" format (default): Full ssh command line.
    $ limactl show-ssh --format=cmd default
    ssh -o IdentityFile="/Users/example/.lima/_config/user" -o User=example -o Hostname=127.0.0.1 -o Port=60022 lima-default

  "args" format: Similar to the cmd format but omits "ssh" and the destination address
    $ limactl show-ssh --format=args default
    -o IdentityFile="/Users/example/.lima/_config/user" -o User=example -o Hostname=127.0.0.1 -o Port=60022

  "options" format: ssh option key value pairs
    $ limactl show-ssh --format=options default
    IdentityFile="/Users/example/.lima/_config/user"
    User=example
    Hostname=127.0.0.1
    Port=60022

  "config" format: ~/.ssh/config format
    $ limactl show-ssh --format=config default
    Host lima-default
      IdentityFile "/Users/example/.lima/_config/user "
      User example
      Hostname 127.0.0.1
      Port 60022


Flags:
  -f, --format string   Format: cmd, args, options, config (default "cmd")
  -h, --help            help for show-ssh

Global Flags:
      --debug   debug mode
```

**Comment by bingdff** [comment]
cmd: limactl show-ssh --format=config default > ./ssh-config
copy the file ssh-config content into the vscode configruation file,  then connect by vscode directly

works well

**Comment by NN---** [comment]
`logrus.Warnf("`limactl show-ssh` is deprecated. Instead, use `ssh -F %s %s`.", filepath.Join(inst.Dir, filenames.SSHConfig), inst.Hostname)`

https://github.com/lima-vm/lima/blob/7685028050802cf24b84762d70017763c5ba6c24/cmd/limactl/show-ssh.go#L89

**Comment by bogdan-panteleev** [comment]
I struggled a lot on how to find the username and the private key, so here is the way how you can find them.
You can get the file as Akihiro recommends [in his comment](https://github.com/lima-vm/lima/discussions/1221#discussioncomment-4283769). There you can see the field 'user' that should be passed to CLion SSH connection interface as 'username'. Also you can see 'IdentityFile', this is the path to the private ssh key

**Reply by afbjorklund** [reply]
That is what the `show-ssh` command did (show you the config)...

Are you sure the IDE does not support using a ssh config file?

**Reply by finalyards** [reply]
IntelliJ IDE's (at least Rust Rover) allow an ssh config file, but only the default `~/.ssh/config`. 

I ended up finding where the SSH config for the particular VM is kept (it's in `~/.lima/{your-vm}/ssh.config`) and `Host *\nInclude`:ing that file in the `~/.ssh/config`. This allows ssh command itself to reach the VM:

```
% ssh lima-den-vm               
me@lima-den-vm:~$ 
```

Oh, and I fixed the SSH port in the YAML. With these settings, entering things into the IDE Remote Development dialog becomes a breeze.

https://github.com/finalyards/Lima-with-RustRover#2-set-up-rust-rover-ide
